prov/efa/src/efa_hmem.c

```diff
@@ -238,6 +238,27 @@ static inline void efa_hmem_info_check_p2p_support_neuron(struct efa_hmem_info *
 	return;
 }
 
+static inline void
+efa_hmem_info_popuplate_iface_string(enum fi_hmem_iface iface, struct efa_hmem_info *info) {
+	switch (iface) {
+		case FI_HMEM_SYSTEM:
+			info->iface_name = "system";
```

Comment:
And you cannot assign the string barely here because it will be freed outside the function. For that you need to use `strdup` to alloc a new mem and free them later. For all of the complexity I don't think you need this function